.github/workflows/maven.yml

```diff
@@ -0,0 +1,27 @@
+# This workflow will build a Java project with Maven
+# For more information see: https://help.github.com/actions/language-and-framework-guides/building-and-testing-java-with-maven
+
+name: Java CI with Maven
+
+# Execute on every pull request and if activated manually.
+on: [pull_request, workflow_dispatch]
```

Comment:
So not on merge to master? Why not?

Reply:
I'm not sure which [workflow trigger event](https://docs.github.com/en/actions/reference/events-that-trigger-workflows#webhook-events) you're referring to here: merging a PR inot master triggers the [`pull_request`](https://docs.github.com/en/actions/reference/events-that-trigger-workflows#pull_request) webhook, which also triggers when a PR is opened, closed, reviewed, and so on. Since I'd like to test PR that merge into other PR (such as PRs that build upon each other), I didn't restrict this to PRs that will be merged with master only.

Do you mean I should add a trigger for all [`push`](https://docs.github.com/en/actions/reference/events-that-trigger-workflows#push) to master? That's not something we do generally, but I could add that if that would be helpful.

Reply:
> merging a PR inot master triggers the [`pull_request`](https://docs.github.com/en/actions/reference/events-that-trigger-workflows#pull_request) webhook

Where do you see that in the docs? Merge isn't even in the activity types, and by default it runs only on `opened`, `synchronize`, or `reopened`. That doesn't even include additional pushes, does it?

Reply:
Ah, good catch. I've added `push` as a trigger in 5924e34f33b7fe5 -- that way, it'll trigger every time a PR is closed by merging it into master, as well as testing every branch I upload even before I've created a PR for it. Do you think that covers all the cases we need, or is there something else I'm missing?